cql3/expr/expression.hh

```diff
@@ -574,6 +574,15 @@ constant evaluate_IN_list(term&, const query_options&);
 cql3::raw_value_view evaluate_to_raw_view(const ::shared_ptr<term>&, const query_options&);
 cql3::raw_value_view evaluate_to_raw_view(term&, const query_options&);
 
+// Takes a prepared expression and calculates its value.
+// Evaluates bound values, calls functions and returns just the bytes and type.
+constant evaluate(const expression& e, const query_options&);
+constant evaluate(const bind_variable&, const query_options&);
+constant evaluate(const tuple_constructor&, const query_options&);
+constant evaluate(const collection_constructor&, const query_options&);
+constant evaluate(const usertype_constructor&, const query_options&);
+constant evaluate(const function_call&, const query_options&);
+
```

Comment:
Do these need to be in a header? The entry point does, but for the subtypes, they'll always be dispatched from the entrypoint, never called directly.

Reply:
I thought it could be useful. Maybe in the future there will be a situation when I have a `collection_constructor` and want to evaluate it (somewhere in prepare?). Without those functions I would have to create an `expression` and call evaluate on it, which is a lot of needless work.